core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -704,6 +704,17 @@ public FileInfo getFileInfo(AlluxioURI path)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException {
     Metrics.GET_FILE_INFO_OPS.inc();
     long flushCounter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
+
+    // Get a READ lock first to see if we need to load metadata, note that this assumes load
```

Comment:
On line 772, we use `LoadMetadataOptions.defaults()`, and for that data structure, the `mLoadDirectChildren` field is `false`.